retrofit/src/main/java/retrofit/Utils.java

```diff
@@ -39,6 +39,12 @@
     return object;
   }
 
+  static boolean methodRequiresRequestBody(final String method) {
```

Comment:
That's not Retrofit's problem. And there's an issue tracking that behavior (https://github.com/square/okhttp/issues/229).